### PR TITLE
Multiple initial PDP announcements [6379]

### PR DESCRIPTION
**Pull request by MiguelCompany**
With this PR a participant will send 1 additional quick announcement (spaced 500ms) when starting. This should improve discovery when several participants are started at the same time.

The user can configure this new feature both on code and XML

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/451/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/409/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/449/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/459/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/417/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/457/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/462/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/419/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/460/)

**Comment by MiguelCompany** [comment]
Liveliness tests seem to be also failing on base branch (1.8.x). See [this manual run](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual/479/)

File: src/cpp/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp [hunk 1]
```diff
@@ -27,52 +27,73 @@
...
+        const BuiltinAttributes& config)
+    : TimedEvent(
+            p_SPDP->getRTPSParticipant()->getEventResource().getIOService(),
+            p_SPDP->getRTPSParticipant()->getEventResource().getThread(),
+            0)
+    , pdp_(p_SPDP)
+    , standard_period_(config.leaseDuration_announcementperiod)
+    , initial_announcements_(config.initial_announcements)
+{
+    if ((initial_announcements_.count > 0) && (initial_announcements_.period <= c_TimeZero))
```
Comment (MiguelBarro) [inline src/cpp/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp]: If the user passes 0 or negative we setup 1ms, maybe we should warn about this.
Note that as tinyxml2 works passing a negative count is actually passing a large positive one.

File: src/cpp/xmlparser/XMLElementParser.cpp [hunk 2]
```diff
@@ -193,6 +200,46 @@ XMLP_ret XMLParser::getXMLBuiltinAttributes(tinyxml2::XMLElement *elem, BuiltinA
...
+    */
+    tinyxml2::XMLElement *p_aux0 = nullptr;
+    const char* name = nullptr;
+    for (p_aux0 = elem->FirstChildElement(); p_aux0 != NULL; p_aux0 = p_aux0->NextSiblingElement())
+    {
+        name = p_aux0->Name();
+        if (strcmp(name, COUNT) == 0)
+        {
+            // portBase - uint16Type
+            if (XMLP_ret::XML_OK != getXMLUint(p_aux0, &config.count, ident))
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/xmlparser/XMLElementParser.cpp]: This should be made on a separate PR

File: src/cpp/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/rtps/builtin/discovery/participant/timedevent/ResendParticipantProxyDataPeriod.cpp]: > maybe we should warn about this

done in commit 43ff2e7254a924debf45ea42d12056a052ca265a

File: include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h [hunk 3]
```diff
@@ -77,6 +77,24 @@ class SimpleEDPAttributes
...
 };
 
+/**
+ * Struct InitialAnnouncementConfig defines the behavior of the RTPSParticipant initial announcements.
+ * @ingroup RTPS_ATTRIBUTES_MODULE
+ */
+struct InitialAnnouncementConfig
+{
+    /// Number of initial announcements with specific period (default 1)
+    uint32_t count = 1u;
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h]: I agree. Changing to 0 on commit 88c8257a34ebef32c243a6bf3fb5dbf27ebee9ba

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/482/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/436/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/479/)

**Pull request merged**